docs/device_support_guide.md

```diff
@@ -163,3 +165,39 @@ To read EC memory in text form run:
 To save EC memory in text form run:
 
 + `sudo dd if=/dev/mem bs=1 skip=4227860480 count=256 | hexdump -C > ec_dump.txt`
+
+# Contribution
+
+## Firmware naming
+
+|                                                      EC                                                       |                                             BIOS                                              |         Series          |
+|:-------------------------------------------------------------------------------------------------------------:|:---------------------------------------------------------------------------------------------:|:-----------------------:|
+| ${\color{red}xxxx}{\color{orange}b}{MS}{\color{yellow}n}{.}{\color{WildStrawberry}y}{\color{YellowOrange}zz}$ | ${E}{\color{red}xxxx}{\color{orange}b}{MS.}{\color{WildStrawberry}y}{\color{YellowOrange}zz}$ | Business/Creator/Gaming |
+| ${\color{red}xxxx}{\color{orange}b}{WS}{\color{yellow}n}{.}{\color{WildStrawberry}y}{\color{YellowOrange}zz}$ | ${E}{\color{red}xxxx}{\color{orange}b}{WS.}{\color{WildStrawberry}y}{\color{YellowOrange}zz}$ | Workstation/CreatorPRO  |
+
++ ${\color{red}xxxx}$ - model code
++ ${\color{orange}b}$ - CPU/EC vendor
+  + For BIOS:
+    + A - AMD CPU
+    + I - Intel CPU
+  + For EC:
+    + E - ENE
+    + I - ITE
++ ${\color{yellow}n}$ - EC / board(?) model
++ ${\color{WildStrawberry}y}{\color{YellowOrange}zz}$ - version
+  + ${\color{WildStrawberry}y}$ - board generation; firmware is incompatible if different
+  + ${\color{YellowOrange}zz}$ - firmware version
+
+## Firmware generations
```

Comment:
i don't see how this might help people discover addresses on their unsupported laptops

Reply:
If people will compare with reference map they need to know how to check about generations